### ISSUE TITLE
TypeArgumentInference: AsSuper crash 

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#2717]
```
public class GenericsAndBounds {

    interface Tree {}

    interface ExpressionTree extends Tree {}

    interface BinaryTree extends ExpressionTree {}

    interface Matcher<T extends Tree> {}


    public static void test(Matcher<? super ExpressionTree> m) {}

    public static void caller() {
        test(toType(BinaryTree.class, GenericsAndBounds.<BinaryTree>allOf()));
    }

    public static <T extends Tree> Matcher<T> allOf() {
        return null;
    }

    public static <S extends T, T extends Tree> Matcher<T> toType(Class<S> type, Matcher<? super S> matcher) {
        return null;
    }
}
```

**Comment by msridhar** [comment]
@smillst was this one fixed by #2736?

**Comment by smillst** [comment]
Yes!  Thanks!
(Fixed via 6c5e4ce)
